coordinator/src/trade/mod.rs

```diff
@@ -516,18 +517,37 @@ impl TradeExecutor {
             leverage_coordinator,
             stable,
         )
+        .await
     }
 
     // Creates a position and a trade from the trade params
-    fn persist_position(
+    async fn persist_position(
         &self,
         connection: &mut PgConnection,
         trade_params: &TradeParams,
         temporary_contract_id: ContractId,
         coordinator_leverage: f32,
         stable: bool,
     ) -> Result<()> {
-        let liquidation_price = liquidation_price(trade_params);
+        let price = trade_params.average_execution_price();
+        let maintenance_margin = { self.node.settings.read().await.maintenance_margin };
```

Comment:
The name suggests that this is an `Amount`, but it's actually a percentage.

Reply:
https://github.com/get10101/10101/pull/2372